### ISSUE TITLE
Mentioning a user not member of space

### ISSUE THREAD

**Issue by habl** [humhub/humhub#3767]
When you mention a user in a comment of message on a space, where that user isn't member of, he won't get notified. But the user will get notified about all the comments made with a link which can't be opened.

### What steps will reproduce the problem?
Mention a user in a space, who is not member of that space. Check notifications, none will be shown. Make another comment, and check notifications again. Now a notification is shown.

### What is the expected result?
Not possible to mention, or no notifications are sent at all.

### What do you get instead?
The didn't got a notification, but user will get notified about all following comments.

### Additional info

| Q                | A
| ---------------- | ---
| HumHub version   | 1.3.17
| PHP version      | 7.0.33
| Operating system | CentOS 7 (humhub running in a docker environment)

**Comment by Ace2213** [comment]
Would be fixed by #3679

**Comment by buddh4** [comment]
Does this only occure for comments private posts?

**Comment by habl** [comment]
Yes if the space is private and a user who is member of the space, mentions a user who isn't he won't get the notification. Probably as intended Indeed. But when more comments are made the user who isn't a member will get notifications from now on, which shouldn't happen I think. Expected behaviour would be to only be able to mention permitted users and without permission never get any notifications.

I hope I'm clear :)

**Comment by habl** [comment]
I think @Ace2213 is right, suggestion would fix the problem.

**Comment by Ace2213** [comment]
I understand why Luke treated it as a bug. It's easier/faster to fix the problem by simply disabling the notifications than to refactor the entire mentioning system.

Facebook handles this scenario by mentioning them (without any notifications), but the mention appears grey on desktop with an additional icon to show that they aren't members and will not see the mention. But this behavior (last I checked) was not applied to their app. In the app it appears like a regular mention, which definitely causes confusion.

I don't think it's useful to mention users that are not members, so it makes more sense to me to prevent mentioning them to begin with. It can also discourage people from talking behind each other's backs 😉

**Comment by habl** [comment]
I agree, limiting the list of users based on where you mention it would be a fine solution and probably easiest way to fix. Thanks for all the thinking so far :)

**Comment by funkycram** [comment]
Yes, this problem occurs regularly : mentioning someone in a private space that is not member of the space. When notified, the user doesn't understand why he cannot see the content.
@habl solution (limiting the list of users to the space members) would be great !

**Comment by buddh4** [comment]
So the problem is rather that the mentioned user is added as subscriber to the content, even though he is not allowed to see the content and gets notification for new comments?
